langchain4j-cassandra/src/main/java/dev/langchain4j/store/embedding/astradb/AstraDbEmbeddingStore.java

```diff
@@ -0,0 +1,294 @@
+     *
+     * @param embeddingList   embeddings
+     * @param textSegmentList text segments
+     * @return list of new row if (same order as the input)
+     */
+    @Override
+    public List<String> addAll(List<Embedding> embeddingList, List<TextSegment> textSegmentList) {
+        if (embeddingList == null || textSegmentList == null || embeddingList.size() != textSegmentList.size()) {
+            throw new IllegalArgumentException("embeddingList and textSegmentList must not be null and have the same size");
+        }
+        // Looping on both list with an index
+        List<String> ids = new ArrayList<>();
+        for (int i = 0; i < embeddingList.size(); i++) {
+            ids.add(add(embeddingList.get(i), textSegmentList.get(i)));
+        }
+        return ids;
+    }
+
+    /** {@inheritDoc}  */
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+        return findRelevant(referenceEmbedding, null, maxResults, minScore);
+    }
+
+    /**
+     * Semantic search with metadata filtering.
+     *
+     * @param referenceEmbedding
+     *      vector
+     * @param metaDatafilter
+     *      fileter for metadata
+     * @param maxResults
+     *      limit
+     * @param minScore
+     *      threshold
+     * @return
+     *      records
+     */
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, Filter metaDatafilter, int maxResults, double minScore) {
+        return collection.similaritySearch(referenceEmbedding.vector(), metaDatafilter,  maxResults)
+                .stream()
+                .filter(r -> r.getSimilarity() >= minScore)
+                .map(this::mapJsonResult)
+                .collect(Collectors.toList());
+    }
+
+    private EmbeddingMatch<TextSegment> mapJsonResult(JsonResult jsonRes) {
+        System.out.println(jsonRes.toString());
+        Double score        = (double) jsonRes.getSimilarity();
+        String embeddingId  = jsonRes.getId();
+        Embedding embedding = Embedding.from(jsonRes.getVector());
+        TextSegment embedded = null;
+        Map<String, Object> properties = jsonRes.getData();
+        if (properties!= null) {
+            String body = properties.getOrDefault(KEY_BODY, "").toString();
+            Metadata metadata = new Metadata(properties.entrySet().stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey,
+                            entry -> entry.getValue() == null ? "" : entry.getValue().toString()
+                    )));
+            // Body is a reserved keyword
+            metadata.remove(KEY_BODY);
+            embedded = new TextSegment(body, metadata);
+
+        }
+        return new EmbeddingMatch<TextSegment>(score, embeddingId, embedding, embedded);
+    }
+
+    /**
+     * Gets collection
+     *
+     * @return value of collection
+     */
+    public CollectionClient getCollectionClient() {
```

Comment:
Please avoid using "get" and "set" in getters and setters (in all classes in the module), we follow modern record style naming.